flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/JsonFunctionsITCase.java

```diff
@@ -782,14 +783,95 @@ private static List<TestSetSpec> jsonSpec() {
                         .testTableApiRuntimeError(
                                 jsonObject(JsonOnNull.NULL, "K", json("{")),
                                 TableRuntimeException.class,
+                                "Invalid JSON string in JSON(value) function: \"{\". Error: org.apache.flink.shaded.jackson2.com.fasterxml.jackson.core.io.JsonEOFException: Unexpected end-of-input: expected close marker for Object (start marker at [Source: (String)\"{\"; line: 1, column: 1])\n"
```

Comment:
Is the entire string necessary? Could we test for `Invalid JSON string in JSON(value) function` only? The rest of the message is rather fragile and can easily change.

Reply:
You actually have a good point and that's a good shout. Adjusted the tests 🙂